DDEyC_API/Controllers/JobListingController.cs

```diff
@@ -1,42 +1,139 @@
     {
-        private readonly IJobListingService _jobListingService;
-        private readonly ILogger<JobListingsController> _logger;
+        private readonly IJSearchService _jSearchService;
+        private readonly ILogger<JobListingController> _logger;
 
-        public JobListingsController(IJobListingService jobListingService, ILogger<JobListingsController> logger)
+        public JobListingController(
+            IJSearchService jSearchService,
+            ILogger<JobListingController> logger)
         {
-
-            _jobListingService = jobListingService;
+            _jSearchService = jSearchService;
             _logger = logger;
         }
+
+        /// <summary>
+        /// Retrieves job listings based on the provided filter criteria
+        /// </summary>
         [HttpGet]
-        public async Task<ActionResult<List<JobListing>>> GetJobListings([FromQuery] JobListingFilter filter)
+        [ProducesResponseType(typeof(List<JobListing>), StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status503ServiceUnavailable)]
+        public async Task<ActionResult<List<JobListing>>> GetJobListings(
+            [FromQuery] JobListingFilter filter,
+            CancellationToken cancellationToken)
         {
-            var jobListings = await _jobListingService.GetJobListingsAsync(filter);
-            return Ok(jobListings);
+            try
+            {
+                _logger.LogInformation(
+                    "Retrieving job listings with filter: {@Filter}",
+                    filter);
+
+                // Initialize filter if null
+                filter ??= new JobListingFilter();
+
+                // Apply default and max limits
+                if (filter.Limit <= 0)
+                {
+                    filter.Limit = 10; // Default limit
+                }
+                else if (filter.Limit > 100)
+                {
+                    filter.Limit = 100; // Maximum limit
+                }
+
+                var jobListings = await _jSearchService.SearchJobListingsAsync(
+                    filter,
+                    cancellationToken);
+
+                _logger.LogInformation(
+                    "Successfully retrieved {Count} job listings",
+                    jobListings.Count);
+
+                return Ok(jobListings);
+            }
+            catch (OperationCanceledException)
+            {
+                _logger.LogInformation("Request was cancelled");
+                return StatusCode(
+                    StatusCodes.Status499ClientClosedRequest,
+                    new ProblemDetails
+                    {
+                        Title = "Request Cancelled",
+                        Detail = "The request was cancelled by the client.",
+                        Status = StatusCodes.Status499ClientClosedRequest
+                    });
+            }
+            catch (JSearchException ex)
```

Comment:
El manejo del bubbling puede mejorarse al permitir que las excepciones generales se propaguen hacia capas superiores en lugar de capturarlas todas aquí. También sería más claro envolver únicamente la llamada a SearchJobListingsAsync en el bloque try-catch, enfocando la gestión de errores donde realmente ocurre. Asegúrate de manejar primero las excepciones específicas como OperationCanceledException antes de capturar excepciones generales. Esto hará el código más limpio y alineado con el principio de bubbling.

Reply:
This is a progressive query builder, it is built on sequential ifs, a switch statement is not adequate for this function, no argument is compared twice or against more than the two values, null or empty, the conditions are not mutually exclusive and no two conditions do the same thing